### PR TITLE
The Great Lambda FixUp

### PR DESCRIPTION
**Pull request by MiniDigger**
Cleans up 125+ conflicting param names for lambda params

### CODE REVIEW

**Comment by MiniDigger** [comment]
fixed some record related issues in [`ad80c98` (#3593)](https://github.com/FabricMC/yarn/pull/3593/commits/ad80c9847268c9a8b3c3fd7aa172ae48f468fbc4), can move to another PR if needed. other than that am happy now, have a compile clean workspace now 🎉

**Comment by liach** [comment]
Did the record inference break, or is it some record accessor-interface inheritance issue?

File: mappings/net/minecraft/server/world/ServerWorld.mapping [hunk 1]
```diff
@@ -243,7 +243,7 @@ CLASS net/minecraft/class_3218 net/minecraft/server/world/ServerWorld
 		ARG 0 writer
 		ARG 1 entities
 	METHOD method_21625 dump (Ljava/nio/file/Path;)V
-		ARG 1 path
+		ARG 1 pathParam # paper
```
Comment (modmuss50) [inline mappings/net/minecraft/server/world/ServerWorld.mapping]:
```suggestion
		ARG 1 pathParam
```

File: mappings/net/minecraft/server/world/ServerWorld.mapping [hunk 2]
```diff
@@ -311,7 +311,7 @@ CLASS net/minecraft/class_3218 net/minecraft/server/world/ServerWorld
 	METHOD method_31418 getLightningRodPos (Lnet/minecraft/class_2338;)Ljava/util/Optional;
 		ARG 1 pos
 	METHOD method_31420 (Lnet/minecraft/class_3695;Lnet/minecraft/class_1297;)V
-		ARG 2 entity
+		ARG 2 entityParam # paper
```
Comment (modmuss50) [inline mappings/net/minecraft/server/world/ServerWorld.mapping]:
```suggestion
		ARG 2 entityParam
```

I dont think there is a reason for these comments?

File: mappings/net/minecraft/server/world/ServerWorld.mapping [hunk 3]
```diff
@@ -332,9 +332,9 @@ CLASS net/minecraft/class_3218 net/minecraft/server/world/ServerWorld
 	METHOD method_33144 isSleepingEnabled ()Z
 		COMMENT {@return whether sleeping can cause the night to skip}
 	METHOD method_34679 (Lnet/minecraft/class_2338;)Lnet/minecraft/class_2338;
-		ARG 0 pos
+		ARG 0 innerPos # yarn
```
Comment (modmuss50) [inline mappings/net/minecraft/server/world/ServerWorld.mapping]:
```suggestion
		ARG 0 innerPos
```

File: mappings/net/minecraft/server/world/ServerWorld.mapping [hunk 4]
```diff
@@ -332,9 +332,9 @@ CLASS net/minecraft/class_3218 net/minecraft/server/world/ServerWorld
...
 	METHOD method_34680 (Lnet/minecraft/class_2338;)Z
-		ARG 1 pos
+		ARG 1 innerPos # yarn
```
Comment (modmuss50) [inline mappings/net/minecraft/server/world/ServerWorld.mapping]:
```suggestion
		ARG 1 innerPos
```

File: mappings/net/minecraft/world/dimension/DimensionType.mapping [hunk 5]
```diff
@@ -1,6 +1,19 @@
 CLASS net/minecraft/class_2874 net/minecraft/world/dimension/DimensionType
+	FIELD comp_641 fixedTime Ljava/util/OptionalLong;
```
Comment (modmuss50) [inline mappings/net/minecraft/world/dimension/DimensionType.mapping]: Why did you need to name these? They should have been automaticaly given these names via https://github.com/FabricMC/name-proposal/

File: mappings/net/minecraft/registry/entry/RegistryEntry.mapping [hunk 6]
```diff
@@ -18,6 +18,7 @@ CLASS net/minecraft/class_6880 net/minecraft/registry/entry/RegistryEntry
 	COMMENT @see RegistryEntry.Reference
 	COMMENT @see net.minecraft.registry.Registry#entryOf
 	COMMENT @see net.minecraft.registry.Registry#getEntry
+	METHOD comp_349 value ()Ljava/lang/Object;
```
Comment (modmuss50) [inline mappings/net/minecraft/registry/entry/RegistryEntry.mapping]: I dont see a related record field mapping for this (I would expect this to be in one of the inner casses). Again this is already auto mapped.

File: mappings/net/minecraft/network/packet/c2s/play/ClientSettingsC2SPacket.mapping [hunk 7]
```diff
@@ -6,6 +6,7 @@ CLASS net/minecraft/class_2803 net/minecraft/network/packet/c2s/play/ClientSetti
 	FIELD comp_270 playerModelBitMask I
 	FIELD comp_271 mainArm Lnet/minecraft/class_1306;
 	FIELD comp_272 filterText Z
+	FIELD comp_273 allowsListing Z
```
Comment (modmuss50) [inline mappings/net/minecraft/network/packet/c2s/play/ClientSettingsC2SPacket.mapping]: As with the other record changes, this should (and is) auto mapped.

**Review by modmuss50 — CHANGES_REQUESTED** [review]
I dont mind adding the x suffix to everything, its not ideal but better than the current clashing names. Ideally we would have the tools to prevent this from happening again in the future.

The record component name changes dont make a lot of sense to me, as they should already have auto mapped names. What was the reasoning behind this, its quite possible something isnt working correct elsewhere.

**Comment by MiniDigger** [comment]
the record stuff I can clean up, hopefully today, it was only a symptom of a bug I had elsewhere in my code, I believe I dont need those changes anymore.

will clean the comments too then

**Comment by MiniDigger** [comment]
that should be all now, stuff still compiles cleanly for me (if you run a jdk that has this stupid bug, that costed me way too much time, fixed... https://bugs.openjdk.org/browse/JDK-8288130), this is read to merge from my point of view

**Review by modmuss50 — APPROVED** [review]
Looks great!

I will allow a little time for anyone (CC: @FabricMC/yarn-triage ) to provide input on specifc lambda names, these renames wont affect modders so im not too worried, we can always improve them later.

👍

**Comment by MiniDigger** [comment]
due to the amount I just gave up trying to name them, you can see that I tried for some of them but didnt want to spend that much time on...

**Review by YanisBft — APPROVED** [review]
As modmuss said, x suffixes are a good temporary solution, so thank you for your time

**Pull request merged**